ProiectPractica/Repository/RepositoryConfigurare.cs

```diff
@@ -5,40 +5,77 @@
 using System.Text;
 using System.Threading.Tasks;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Schema;
+using Newtonsoft.Json.Linq;
 
 namespace ProiectPractica.Repository
 {
-    internal class RepositoryConfigurare
+    internal class RepositoryConfigurare : IRepository
     {
         /// <summary>
-        /// this function reads from a file and returns the string read from the file, if the file exists
+        ///  This function reads from a file and returns the string read from the file, if the file exists.
+        ///  And if the file does not exists the function throws an exception.
         /// </summary>
         /// <param name="filePath"></param>
-        /// <returns>a string</returns>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
         private string ReadFromFile(string filePath)
         {
-            var fileInfo = new FileInfo(filePath);
-            if (!fileInfo.Exists)
+            if (!File.Exists(filePath))
             {
-                return "";
+                throw new Exception("File not found!");
             }
             return File.ReadAllText(filePath);
         }
+
         /// <summary>
-        /// it takes the string from ReadFromFile,and if is not empty, it makes a Configurare object, deserializing the string 
-        /// if it is empty it returns an empty object Configurare
+        /// This function deserializes a JSON string if the string is valid and returns the configuration.
+        /// If the string fails the validation or the file does not exist it returns an empty object
         /// </summary>
         /// <param name="filePath"></param>
-        /// <returns>Configurare</returns>
-        public Configurare GetConfugurareFromString(string filePath)
+        /// <returns>The configuration from the file if it is valid or an empty Configurare object otherwise</returns>
+        public Configurare GetConfigurareFromString(string filePath)
         {
-            string stringFromFile = ReadFromFile(filePath);
-            if(stringFromFile == "")
+            try
+            {
+                var stringFromFile = ReadFromFile(filePath);
+                if (ValidateString(stringFromFile))
+                {
+                     var configurare = JsonConvert.DeserializeObject<Configurare>(stringFromFile);
+                     return configurare;
+                }
+                return new Configurare();
+            }
+            catch (Exception ex)
             {
+                Console.WriteLine(ex.Message);// sa adaug un logger in cazul in care am erori de genul?
```

Comment:
Oare ar fi mai ok sa am un logger pentru erori, sau cum recomanzi sa urmaresc erorile?

Reply:
In mod normal am folosi un logger, dar nu cred ca e necesar momentan. 
In orice caz, nu ma asteptam sa tratezi exceptiile in metoda asta. E ok ca metoda doar sa arunce exceptii daca ceva nu este in regula. 

In general, exceptiile nu sunt folosite ca sa putem face business logic pe baza lor. Din punctul meu de vedere, daca fisierul de configurare nu poate fi citit, nu prea are sens sa putem merge mai departe, sa decidem sa returnam un obiect de configurare gol. As recomanda doar sa aruncam exceptii, si sa le prindem eventual in Main. Putem sa mai discutam daca vrei.